woocommerce/woocommerce-gutenberg-products-block
# Move ProductSortSelect and ProductListItem under /product-list

Part of #1962.

This PR moves the `product-sort-select` & `product-list-item` folders under `product-list`, as they are used only by the ProductList component. It will close  #1962 completely, as it is the last sub-task.

What is done in this PR:
- `/base/components/product-sort-select` is now at  `/base/components/product-list/product-sort-select`
- `/base/components/product-list-item` is now at `/base/components/product-list/product-list-item`

### How to test the changes in this Pull Request:

1. Verify the new folder structure and whether it makes sense.
2. Make sure the project still builds and do some smoke testing to verify All products block continues working.
